src/OpenTelemetryEngine/Logging/FunctionsResourceDetector.cs

```diff
@@ -0,0 +1,80 @@
+//
+using System.Diagnostics;
+using OpenTelemetry.Resources;
+using OpenTelemetryEngine.Constants;
+
+namespace OpenTelemetryEngine.Logging
+{
+internal sealed class FunctionsResourceDetector : IResourceDetector
+    {
+        public Resource Detect()
+        {
+            List<KeyValuePair<string, object>> attributeList = new(9);
+            try
+            {
+                string? serviceName = Environment.GetEnvironmentVariable(ResourceAttributeConstants.SiteNameEnvVar);
+                // Todo: This is probably wrong, but we don't have access to types from the worker in this context
+                // We may have to pass the worker's assembly version manually in setup, will figure this out later
+                string? version = typeof(FunctionsResourceDetector).Assembly.GetName()?.Version?.ToString();
+
+                if (!string.IsNullOrEmpty(version)) 
+                {
+                    attributeList.Add(new KeyValuePair<string, object>(ResourceAttributeConstants.ServiceVersion, version));                    
+                }
+
+                attributeList.Add(new KeyValuePair<string, object>(ResourceAttributeConstants.AISDKPrefix, $@"{ResourceAttributeConstants.SDKPrefix}:{version}"));
+                attributeList.Add(new KeyValuePair<string, object>(ResourceAttributeConstants.ProcessId, Process.GetCurrentProcess().Id));
+
+                // Add these attributes only if running in Azure.
+                if (!string.IsNullOrEmpty(serviceName))
+                {
+                    attributeList.Add(new KeyValuePair<string, object>(ResourceAttributeConstants.ServiceName, serviceName));
+                    attributeList.Add(new KeyValuePair<string, object>(ResourceAttributeConstants.CloudProvider, ResourceAttributeConstants.AzureCloudProviderValue));
+                    attributeList.Add(new KeyValuePair<string, object>(ResourceAttributeConstants.CloudPlatform, ResourceAttributeConstants.AzurePlatformValue));
+
+                    if (!string.IsNullOrEmpty(version))
+                    {
+                        attributeList.Add(new KeyValuePair<string, object>(ResourceAttributeConstants.FaaSVersion, version));
+                    }
+
+                    string? region = Environment.GetEnvironmentVariable(ResourceAttributeConstants.RegionNameEnvVar);
+                    if (!string.IsNullOrEmpty(region))
+                    {
+                        attributeList.Add(new KeyValuePair<string, object>(ResourceAttributeConstants.CloudRegion, region));
+                    }
+
+                    var azureResourceUri = GetAzureResourceURI(serviceName);
+                    if (azureResourceUri != null)
+                    {
+                        attributeList.Add(new KeyValuePair<string, object>(ResourceAttributeConstants.CloudResourceId, azureResourceUri));
+                    }
+                }
+            }
+            catch
```

Comment:
Swallowing all exceptions without rethrowing or logging is concerning. What if this is an OutOfMemoryException? If this code is ultimately going to be called by the OTel SDK, perhaps we shouldn't even try to catch exceptions and let the invoker deal with them? Or, if there is a limited set of specific reasons why we expect an exception that's ok to ignore, we should catch just those specific exceptions.

Reply:
Discussed this with Rohit - I have added a log line to warn if the detector initialization fails, but it is better to continue with the detector in this state than to fail outright. 
Eventually, this detector will move to the otel contrib repo and be shared between the host, .NET worker, and PS module. At that point, we can make further decisions about the execution flow.